backend/src/main/kotlin/app/ehrenamtskarte/backend/application/webservice/schema/view/ApplicationView.kt

```diff
@@ -14,33 +15,42 @@ data class ApplicationView(
     val jsonValue: String,
     val withdrawalDate: String?,
     val note: String?,
+    @GraphQLDeprecated("Use 'status' instead")
     val cardCreated: Boolean?,
+    val status: ApplicationStatus?,
+    val statusResolvedDate: String? = null,
 ) {
     companion object {
-        fun fromDbEntity(entity: ApplicationEntity, includePrivateInformation: Boolean = false): ApplicationView {
-            if (includePrivateInformation) {
-                return ApplicationView(
-                    entity.id.value,
-                    entity.regionId.value,
-                    entity.createdDate.toString(),
-                    entity.jsonValue,
-                    entity.withdrawalDate?.toString(),
-                    entity.note,
-                    entity.cardCreated,
-                )
-            }
-            return ApplicationView(
-                entity.id.value,
-                entity.regionId.value,
-                entity.createdDate.toString(),
-                entity.jsonValue,
-                entity.withdrawalDate?.toString(),
-                null,
-                false,
+        fun fromDbEntity(entity: ApplicationEntity, includePrivateInformation: Boolean = false): ApplicationView =
+            ApplicationView(
+                id = entity.id.value,
+                regionId = entity.regionId.value,
+                createdDate = entity.createdDate.toString(),
+                jsonValue = entity.jsonValue,
+                withdrawalDate = entity.withdrawalDate?.toString(),
+                note = entity.note.takeIf { includePrivateInformation },
+                cardCreated = (entity.status == ApplicationEntity.Status.ApprovedCardCreated)
```

Comment:
❓ Can you explain me what this `includePrivateInformation` is for?

Reply:
From my understanding, this view (should actually be named type IMHO) is reused on several queries, which may or may not include sensitive fields.

I would have created two separate types here, but kept the logic as is for now.